src/components/formComponents/roundsArray.tsx

```diff
@@ -0,0 +1,76 @@
+import React from "react";
+import { Workout } from "../../types";
+import { FieldArray } from "formik";
+import {
+  sectionStyle,
+  sectionTitleStyle,
+  RemoveButton,
+  Select,
+  Input,
+  AddButton
+} from ".";
+import { intervalTypeOptions, newWorkset, newRound } from "../../data";
+import WorksetsArray from "./worksetsArray";
+
+type RoundsArrayProps = {
+  values: Workout;
+  workgroupIndex: number;
+};
+export const RoundsArray: React.FC<RoundsArrayProps> = ({
+  values,
+  workgroupIndex
+}) => (
+  <FieldArray
+    name="rounds"
+    render={roundsArrayHelpers => (
+      <div>
+        {values.workgroups[workgroupIndex].rounds.map((_round, roundIndex) => (
+          <div
+            className={sectionStyle}
+            key={`wg${workgroupIndex}rd${roundIndex}`}
+          >
+            <div className="flex justify-between">
+              <div className={sectionTitleStyle}>
+                {`Round ${roundIndex + 1}`}
+              </div>
+              <RemoveButton
+                remove={roundsArrayHelpers.remove}
+                text="remove"
+                textColor="gray"
+                bgColor="red"
+                hoverColor="red"
+                index={roundIndex}
+              />
+            </div>
+            <div className="flex">
+              <Select
+                labelText="Interval Type"
+                options={intervalTypeOptions}
+                fieldName={`workgroups.${workgroupIndex}.rounds.${roundIndex}.intervalType`}
```

Comment:
@DeathstarNovember - have you tried `workgroups[${workgroupIndex}].rounds[${roundIndex}].intervalType`? I'm not sure whether the array indexing vs. object indexing makes a difference here but it's the first thing I'd try.

Reply:
See https://jaredpalmer.com/formik/docs/guides/arrays#arrays